src/components/book/BookForm.jsx

```diff
@@ -1,15 +1,38 @@
-const BookForm = () => (
-  <div>
-    <h3>ADD NEW BOOK</h3>
-    <form action="">
-      <input type="text" placeholder="Book title" />
-      <select>
-        <option value="action">Action</option>
-        <option value="action">Science Fiction</option>
-      </select>
-      <button type="submit">ADD BOOK</button>
-    </form>
-  </div>
-);
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../../redux/books/books';
+
+const BookForm = () => {
+  const dispatch = useDispatch();
+
+  const addBookToStore = (e) => {
+    e.preventDefault();
+    const title = document.getElementById('title').value;
+    const author = document.getElementById('author').value;
+    const newBook = {
+      id: uuidv4(),
+      title,
+      author,
+    };
+    dispatch(addBook(newBook));
+    document.getElementById('title').value = '';
+    document.getElementById('author').value = '';
+  };
```

Comment:
- [ ] Kindly, update this component. It should contain a local state with `title` and `author`. The state should be updated using the (`onChange`) listener on each input field.
- [ ] Submitting an item to the store should take the input from the component's local state, then invoke `dispatch(addBook(newBook))`.

Reply:
Hello @omar-labana,
I don't understand this part. Local state should empty. I had it then I delete it after see requirement. Also why I need `onChange` listener. This is working with `addBookStore` function. Function  is adding to store correctly.

Reply:
Hi @ahmetbozaci,
You should save any input data by the user in a local state using `useState` react hook, it's the standard for second-tier form validation that you may actually use later, it's also one of the requirements of the project.
If you need more info, kindly reach out on slack.